### PR TITLE
Fix counts datatables

### LINKED ISSUE #466: Make search affect counts on datatables
"I see I can search for a given test on the bottom, but could I have this to the entire tab? So it would also update the statistics on the top?"

Image

Also, when changing between All/Success/Failed/Inconclusive, the search field should be applied automatically (no need to type again)

### PR DESCRIPTION
**Pull request by Francisco2002**
- make filters affect counts in datatables header

Close #466

### CODE REVIEW

File: dashboard/src/components/BootsTable/BootsTable.tsx [hunk 1]
```diff
@@ -285,7 +292,11 @@ export function BootsTable({
       <BaseTable headerComponents={tableHeaders}>
         <TableBody>{tableRows}</TableBody>
       </BaseTable>
-      <PaginationInfo table={table} data={data} intlLabel="treeDetails.boots" />
+      <PaginationInfo
+        table={table}
+        data={table.getPrePaginationRowModel().rows.map(row => row.original)}
```
[the remark this answers is not in the document]
Comment (Francisco2002) [reply dashboard/src/components/BootsTable/BootsTable.tsx]: `rawData.tests` will pass all the data from the pagination component, but I need the filtered data only. But I can get the table props of the component to get this information, so, data props is unnecessary and I remove it.

File: dashboard/src/components/BuildsTable/BuildsTable.tsx [hunk 2]
```diff
@@ -297,14 +305,18 @@ export function BuildsTable({
...
+                : ''
+            }
             navigationLogsActions={navigationLogsActions}
           />
         </Sheet>
       </BaseTable>
       <PaginationInfo
         table={table}
-        data={data}
+        data={table.getPrePaginationRowModel().rows.map(row => row.original)}
```
[the remark this answers is not in the document]
Comment (Francisco2002) [reply dashboard/src/components/BuildsTable/BuildsTable.tsx]: ditto about unnecessary prop

File: dashboard/src/components/Table/PaginationInfo.tsx [hunk 3]
```diff
@@ -33,28 +35,26 @@ interface IPaginationInfo {
...
   intlLabel,
 }: IPaginationInfo): JSX.Element {
   const buttonsClassName = 'text-blue font-bold';
 
+  const filteredRowModel = table.getFilteredRowModel();
+  const countData = useMemo(
+    () => filteredRowModel.rows.length,
+    [filteredRowModel],
+  );
+
```
Comment (MarceloRobert) [inline dashboard/src/components/Table/PaginationInfo.tsx:52]: you can skip filteredRowModel and get the countData _length_ directly, so you can also use this in the span below at line 67

Comment (Francisco2002) [reply dashboard/src/components/Table/PaginationInfo.tsx:52]: if I understood correctly, your suggestion is to do something like this

Comment (Francisco2002) [reply dashboard/src/components/Table/PaginationInfo.tsx:52]: If I understood correctly, your suggestion is to do something like this

`const countData = useMemo(
    () => table.getFilteredRowModel().rows.length,
    [table],
  );`

This way don't update the page count correctly

Comment (MarceloRobert) [reply dashboard/src/components/Table/PaginationInfo.tsx:52]: As we discussed, it makes sense to do how you're doing, it is also done that way in other places. Also, my mistake, I confused countData with the old data.length, the way you're doing is right.

**Review by MarceloRobert — APPROVED** [review]
Great work

**Pull request merged**